tests/perf/test_engineer_traction.py

```diff
@@ -0,0 +1,283 @@
+        engineer_traction()
+
+    with betamax_recorder.use_cassette('awsy-JS', serialize_with='prettyjson'):
+        try:
+            engineer_traction('awsy', 'JS')
+        except TypeError:
+            pytest.fail()
+
+
+def test_formula_exposes_oldest_timestamp(session):
+    engineer_traction = EngineerTractionFormula(session)
+    no_older_than = datetime.now() - timedelta(weeks=24, seconds=5)
+
+    assert engineer_traction.oldest_timestamp >= no_older_than
+
+
+@pytest.mark.parametrize(
+    'cooled_down_bug',
+    [
+        {"creation_time": "2020-05-18T15:20:55Z"},  # older than 2 weeks
+        {"creation_time": "2020-05-04T15:20:55Z"},  # older than 1 month
+        {"creation_time": "2019-12-16T08:10:37Z"},  # older than 6 months
+    ],
+)
+def test_formula_correctly_detects_cooled_down_bugs(cooled_down_bug, session):
+    engineer_traction = EngineerTractionFormula(session)
+
+    assert engineer_traction.has_cooled_down(cooled_down_bug)
+
+
+@pytest.mark.parametrize(
+    'not_cooled_down_bug',
+    [
+        {'creation_time': '2020-05-31T00:00:00Z'},  # 2 days old
+        {'creation_time': '2020-05-26T00:00:00Z'},  # 1 week old
+        {'creation_time': '2020-05-19T23:00:00Z'},  # ~2 weeks old, except for 1 hour
+    ],
+)
+def test_formula_detects_bugs_that_didnt_cool_down_yet(not_cooled_down_bug, session):
+    engineer_traction = EngineerTractionFormula(session)
+
+    assert not engineer_traction.has_cooled_down(not_cooled_down_bug)
+
+
+@pytest.mark.parametrize('bad_structured_bug', [{}, {'creation_time': 'jiberish-date'}])
+def test_formula_throws_adequate_error_for_bug(bad_structured_bug, session):
+    engineer_traction = EngineerTractionFormula(session)
+
+    with pytest.raises(ValueError):
+        engineer_traction.has_cooled_down(bad_structured_bug)
+
+
+def test_accessing_breakdown_without_prior_calculus_errors_out(session):
+    engineer_traction = EngineerTractionFormula(session)
+
+    with pytest.raises(RuntimeError):
+        _ = engineer_traction.breakdown()
+
+
+# Leveraging HTTP VCR
+
+
+def test_breakdown_updates_between_calculations(betamax_recorder):
+    engineer_traction = EngineerTractionFormula(betamax_recorder.session)
+
+    test_moniker_A = ('build_metrics', 'build times')
+    test_moniker_B = ('talos', 'tp5n', 'nonmain_startup_fileio')
+
+    cassette_preffix_A = '-'.join(filter(None, test_moniker_A))
+    cassette_preffix_B = '-'.join(filter(None, test_moniker_B))
+
+    with betamax_recorder.use_cassette(f'{cassette_preffix_A}', serialize_with='prettyjson'):
+        engineer_traction(*test_moniker_A)  # let it perform calculus & cache breakdown
+        breakdown_A = engineer_traction.breakdown()
+
+    with betamax_recorder.use_cassette(f'{cassette_preffix_B}', serialize_with='prettyjson'):
+        engineer_traction(*test_moniker_B)  # let it perform calculus & cache breakdown
+        breakdown_B = engineer_traction.breakdown()
+
+    assert breakdown_A != breakdown_B
```

Comment:
Test that require more control over time-based APIs *(VCR is assuming we're always on June 2nd, 2020)*

Reply:
@ionutgoldan where is the artifact that tells VCR it is June 2nd, 2020?

Reply:
I've setup this global var, in the test module:
```python
CASSETTE_LIBRARY_DIR = 'tests/sample_data/betamax_cassettes/perf_sheriffing_criteria'
```
So I suggest we store all recordings in `tests/sample_data/betamax_cassettes` & organize them there in sub dirs, depending on the feature/component/module that's tested.